pkg/controllers/nodeclaim/disruption/drift.go

```diff
@@ -25,14 +25,17 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
+	corev1 "k8s.io/api/core/v1"
+
 	v1 "sigs.k8s.io/karpenter/pkg/apis/v1"
 	"sigs.k8s.io/karpenter/pkg/cloudprovider"
 	"sigs.k8s.io/karpenter/pkg/scheduling"
 )
 
 const (
-	NodePoolDrifted     cloudprovider.DriftReason = "NodePoolDrifted"
-	RequirementsDrifted cloudprovider.DriftReason = "RequirementsDrifted"
+	NodePoolDrifted        cloudprovider.DriftReason = "NodePoolDrifted"
+	RequirementsDrifted    cloudprovider.DriftReason = "RequirementsDrifted"
+	StaleInstanceTypeDrift cloudprovider.DriftReason = "StaleInstanceTypeDrifted"
```

Comment:
```suggestion
	StaleInstanceTypeDrift cloudprovider.DriftReason = "InstanceTypeNotFound"/"InstanceTypeUnavailable"
```

There was no drift here so I wonder if the reason should contain "Drifted in it"